### PR TITLE
Autofill improvements to onboarding for device auth

### PR DESCRIPTION
**Pull request by amddg44**
Task/Issue URL: https://app.asana.com/0/1201462886803403/1203328956136997/f
Tech Design URL:
CC: @THISISDINOSAUR 

**Description**:
When a user does not have any saved logins, allow them to access the Autofill Logins screen + create / view / edit a new login without requiring authentication. Note that this is subject to outcome of security triage review

**Steps to test this PR**:
1. From a clean install, access the Autofill Logins screen and confirm you are not prompted to authenticate
2. Add a new login, segue back to the Autofill Logins list and confirm you are not prompted to authenticate
3. Tap to view the newly created login and confirm you are not prompted to authenticate
4. Back out of the Autofill Logins screen, then go back in. You should now be prompted to grant biometrics permission and authenticate before you can access the saved logins
5. As a regression test, delete any saved logins and then go to a website and login. Save the login when prompted and tap to view it. Confirm you are prompted to authenticate in order to view it

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

**Review by THISISDINOSAUR — COMMENTED** [review]
After step 3, I went back to the main autofill screen and backgrounded the app, and wasn't prompted to authenticate, but probably should be?
Otherwise looks good!

**Comment by amddg44** [comment]
Great catch! I've added in handling for the main autofill + details screens for when app is backgrounded, thanks!

File: DuckDuckGo/AutofillLoginDetailsViewController.swift [hunk 1]
```diff
@@ -52,15 +52,19 @@ class AutofillLoginDetailsViewController: UIViewController {
...
         self.authenticator = authenticator
         self.authenticationNotRequired = authenticationNotRequired
         super.init(nibName: nil, bundle: nil)
         self.viewModel.delegate = self
 
     }
-    
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
```
Comment (THISISDINOSAUR) [inline DuckDuckGo/AutofillLoginDetailsViewController.swift]: I think we don't need to explicitly call removeObserver on deinit anymore, so would be inclined to remove (although perhaps should put it to the team)

**Review by THISISDINOSAUR — APPROVED** [review]
LGTM! Backgrounding seems to work as expected now

File: DuckDuckGo/AutofillLoginDetailsViewController.swift (hunk 1 above)
Comment (amddg44) [reply DuckDuckGo/AutofillLoginDetailsViewController.swift]: Good catch, thanks! That's removed (I'd forgotten to remove - it was a hangover from some earlier testing when I thought that NotificationCenter observer wasn't being released)

**Pull request merged**